### PR TITLE
Clone owner_id to group_id in miq_sets table for dashboards assigned to group

### PR DESCRIPTION
**Pull request by yrudman**
This PR should go together with https://github.com/ManageIQ/manageiq/pull/19491

Fixes issue https://github.com/ManageIQ/manageiq/issues/18924

### CODE REVIEW

File: db/migrate/20191112132759_copy_owner_id_to_group_id_for_dashboards.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class CopyOwnerIdToGroupIdForDashboards < ActiveRecord::Migration[5.1]
+  class MiqSet < ActiveRecord::Base; end
+
+  def up
+    say_with_time("Copying owner_id to group_id in miq_sets table for each dashboard assigned to group") do
+      MiqSet.where(:set_type => "MiqWidgetSet", :owner_type => "MiqGroup").find_each do |dashboard|
+        dashboard.update(:group_id => dashboard.owner_id)
+      end
```
Comment (Fryguy) [inline db/migrate/20191112132759_copy_owner_id_to_group_id_for_dashboards.rb]:
```ruby
MiqSet.where(:set_type => "MiqWidgetSet", :owner_type => "MiqGroup").update_all("owner_id = group_id")
```

File: db/migrate/20191112132759_copy_owner_id_to_group_id_for_dashboards.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    end
+  end
+
+  def down
+    say_with_time("Nullifying group_id column in miw_sets table for each dashboard assigned to group") do
+      MiqSet.where(:set_type => "MiqWidgetSet", :owner_type => "MiqGroup").find_each do |dashboard|
+        dashboard.update(:group_id => nil)
```
Comment (Fryguy) [inline db/migrate/20191112132759_copy_owner_id_to_group_id_for_dashboards.rb]:
```ruby
MiqSet.where(:set_type => "MiqWidgetSet", :owner_type => "MiqGroup").update_all(:group_id => nil)
```

File: spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require_migration
+
+describe CopyOwnerIdToGroupIdForDashboards do
+  let(:miq_set)  { migration_stub(:MiqSet) }
+  let(:name)     { "Hello Dashboard" }
+  let(:group_id) { 1 }
+
+  migration_context :up do
+    it "Copying owner_id to group_id in miq_sets table for each dashboard assigned to group only" do
+      dashboard = miq_set.create!(:name => name, :set_type => "MiqWidgetSet", :owner_type => "MiqGroup", :owner_id => group_id)
```
Comment (Fryguy) [inline spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb]: Can you also create examples of non-MiqWidgetSet rows, and ensure they are not modified?

File: spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+      migrate
+
+      expect(dashboard.reload.group_id).to eq(group_id)
+    end
+  end
+
+  migration_context :down do
+    it "Nullifying group_id column in miw_sets table for each dashboard assigned to group" do
+      dashboard = miq_set.create!(:name => name, :set_type => "MiqWidgetSet", :owner_type => "MiqGroup", :owner_id => group_id, :group_id => group_id)
```
Comment (Fryguy) [inline spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb]: Ditto

File: db/migrate/20191112132759_copy_owner_id_to_group_id_for_dashboards.rb (hunk 1 above)
Comment (yrudman) [reply db/migrate/20191112132759_copy_owner_id_to_group_id_for_dashboards.rb]: 👍

File: db/migrate/20191112132759_copy_owner_id_to_group_id_for_dashboards.rb (hunk 2 above)
Comment (yrudman) [reply db/migrate/20191112132759_copy_owner_id_to_group_id_for_dashboards.rb]: 👍

File: spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb (hunk 3 above)
Comment (yrudman) [reply spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb]: ok

File: spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
+require_migration
+
+describe CopyOwnerIdToGroupIdForDashboards do
+  let(:miq_set)          { migration_stub(:MiqSet) }
+  let(:name)             { "Hello Dashboard" }
+  let(:group_id)         { 1 }
```
Comment (Fryguy) [inline spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb]: Don't hardcode the group_id as this is invalid in a multi-region mode.  I believe there are specs where we need to create ids from an appropriate region, so search the specs for that...something around ActiveRecord::IdRegions.

Comment (Fryguy) [reply spec/migrations/20191112132759_copy_owner_id_to_group_id_for_dashboards_spec.rb]: As agreed, we'll keep this as is, as it seems we don't have an existing mechanism for generating regionalized ids.

**Pull request merged**